onnx/defs/math/defs.cc

```diff
@@ -1029,6 +1029,48 @@ ONNX_OPERATOR_SET_SCHEMA(
             "Constrain input and output types to float tensors.")
         .TypeAndShapeInferenceFunction(propagateShapeAndTypeFromFirstInput));
 
+static const char* HardSwish_ver14_doc = R"DOC(
+HardSwish takes one input data (Tensor<T>) and produces one output data (Tensor<T>) where
+the HardSwish function, y = x * max(0, min(1, alpha * x + beta)) = x * HardSigmoid<alpha, beta>(x),
+where alpha = 1/6 and beta = 0.5, is applied to the tensor elementwise.
+)DOC";
+
+ONNX_OPERATOR_SET_SCHEMA(
+    HardSwish,
+    14,
+    OpSchema()
+        .SetDoc(HardSwish_ver14_doc)
+        .Input(
+            0,
+            "X",
+            "Input tensor",
+            "T",
+            OpSchema::Single,
+            true,
+            1,
+            OpSchema::Differentiable)
+        .Output(
+            0,
+            "Y",
+            "Output tensor",
+            "T",
+            OpSchema::Single,
+            true,
+            1,
+            OpSchema::Differentiable)
+        .TypeConstraint(
+            "T",
+            {"tensor(float16)", "tensor(float)", "tensor(double)"},
+            "Constrain input and output types to float tensors.")
+        .TypeAndShapeInferenceFunction(propagateShapeAndTypeFromFirstInput)
+        .FunctionBody(FunctionBodyHelper::BuildNodes({
+            // nodes: {outputs, op, inputs, attributes}
+            {{"HS_X"},
+             "HardSigmoid",
+             {"X"},
+             {MakeAttribute("alpha", float(1/6)), MakeAttribute("beta", 0.5f)}},
```

Comment:
I think ```float(1/6)``` evaluates to ```0``` in C++? Could change it to ```1.0f/6.0f```.

Reply:
Is there some test to validate this? Or catch any such issue?

Reply:
What I mean is: the "expanded" test-cases generated will help test the function-body. Running ORT or some other backend with those expanded test-cases would help validate correctness of the function body. (Unfortunately, I don't think this can be done in CI currently.)

Reply:
I fixed the bug in type casting.
I'm not aware of any tests which are run on ORT to check correctness of functions' body.
However, I ran the expanded model with the bugfix on ORT by myself and it returned correct output. 
(One thing that I had to change was the opset version. Currently,  ORT provides  support for domain ai.onnx till opset 13.)